sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -191,22 +196,56 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
     cancelledDate: Date,
     newIssuingProvince: string,
   ): Promise<UpdateResult> {
-    if (!cancelledDate || !newIssuingProvince) {
-      throw new Error(
-        "Not all required fields to update a received MSFAA record were provided.",
+    return this.dataSource.transaction(async () => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      if (!cancelledDate || !newIssuingProvince) {
+        throw new Error(
+          "Not all required fields to update a received MSFAA record were provided.",
+        );
+      }
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
```

Comment:
May not be part of the PR, but we are not throwing error here on `(!updateResult.affected)` similar to `updateReceivedRecord()`. It will be nice to add that.

Reply:
Added it. Thanks 👍